File: 구은성/9월/0926/12943.java
class Solution {
    public int solution(long num) {
        int answer = 0;
        
        if (num == 1) {
            answer = 0;
        }
        else {
            answer = collatz(num, 0);
        }
        return answer;
    }
    
    public int collatz(long num, int cnt) {
        if (num % 2 == 0) {
            num = num / 2;
        }
        else {
            num = num * 3 + 1;
        }
        
        cnt++;
        
        if (num == 1){
            return cnt;
        }
        else if (cnt >= 500) {
            return -1;
        }
        else {
            return collatz(num, cnt);
        }
    }
}

File: 구명준/10월/1002/12943.py
def solution(num):
    cnt = 0
    answer = 0
    while True:
        if num == 1:
            break
        if cnt == 500:
            cnt = -1
            break
        if num%2==0:
            num = num/2
            cnt +=1
        elif num%2 !=0:
            num = (num*3)+1
            cnt +=1
    answer = cnt
    return answer